python/examples/advanced_agents/indie_hacker_agent/langgraph/main.py

```diff
@@ -0,0 +1,75 @@
+from composio import Composio, ComposioToolSet
+import os
+from dotenv import load_dotenv
+from langchain_groq import ChatGroq
+from langchain_openai import ChatOpenAI
+from langgraph.checkpoint.memory import MemorySaver
+from typing import Annotated
+from langchain_core.messages import BaseMessage
+from typing_extensions import TypedDict
+from composio_langgraph import Action, ComposioToolSet, App
+from langgraph.graph import StateGraph, START, END
+from langgraph.graph.message import add_messages
+from langgraph.prebuilt import ToolNode, tools_condition
+load_dotenv()
+
+class State(TypedDict):
+    messages: Annotated[list, add_messages]
+
+graph_builder = StateGraph(State)
+
+toolset = ComposioToolSet()
+tools = toolset.get_tools(apps=[App.CODEINTERPRETER, App.FILETOOL, App.HACKERNEWS, App.SHELLTOOL, App.WEBTOOL])
+
+llm = ChatOpenAI(model="gpt-4o")
+#llm = ChatGroq(model='llama3.3-70b-versatile')
+llm_with_tools = llm.bind_tools(tools)
+
+def chatbot(state: State):
+    return {"messages": [llm_with_tools.invoke(state["messages"])]}
+
+graph_builder.add_node("chatbot", chatbot)
+
+tool_node = ToolNode(tools=tools)
+graph_builder.add_node("tools", tool_node)
+
+graph_builder.add_conditional_edges(
+    "chatbot",
+    tools_condition,
+)
+# Any time a tool is called, we return to the chatbot to decide the next step
+graph_builder.add_edge("tools", "chatbot")
+graph_builder.add_edge(START, "chatbot")
+memory = MemorySaver()
+
+graph = graph_builder.compile(checkpointer=memory)
+
+config = {"configurable": {"thread_id": "1"}, "recursion_limit": 50}
```

Comment:
The thread_id in the config is hardcoded as "1". Consider making this configurable or generating a unique ID to avoid potential conflicts in concurrent usage.